src/io/EngineLogAccess.cpp

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#include "EngineLogAccess.hpp"
+#include "io/FileLineReader.hpp"
+#include "io/KeyValueFileReader.hpp"
+#include "io/KeyValueFileWriter.hpp"
+#include "LocalPath.hpp"
+
+unsigned
+EngineLogAccess::ReadLog()
+{
+  try{
+    FileLineReaderA file(LocalPath(log_file_name));
+    KeyValueFileReader kvreader(file);
+    KeyValuePair pair;
+    while (kvreader.Read(pair)) {
+      std::string bt_mac(pair.key);
+      double engine_runtime = std::atof(pair.value);
+      SensorToLog(bt_mac, engine_runtime);
+    }
+  } catch (...){}
+  return engine_log.size();
+}
+
+bool
+EngineLogAccess::WriteLog()
+{
+  try{
+    FileOutputStream file(LocalPath(log_file_name), Mode::CREATE);
+    BufferedOutputStream buffered(file);
+    KeyValueFileWriter kvwriter(buffered);
+
+    for (auto const& elem : engine_log) {    
+      kvwriter.Write(elem.first.c_str(), std::to_string(elem.second).c_str());    
+    }
+
+    buffered.Flush();
+    file.Commit();
+  } catch (...){
+    return false;
+  }
+  return true;
+}
+
+bool
+EngineLogAccess::WriteSingleUpdateToLog(std::string bt_mac, double engine_runtime)
+{
+  SensorToLog(bt_mac, engine_runtime);
+
+  try{
+    FileOutputStream file(LocalPath(log_file_name), Mode::APPEND_OR_CREATE);
+    BufferedOutputStream buffered(file);
+    KeyValueFileWriter kvwriter(buffered);
+    kvwriter.Write(bt_mac.c_str(), std::to_string(engine_runtime).c_str());    
+    buffered.Flush();
+    file.Commit();
+  } catch (...){
+    return false;
```

Comment:
I don't like suppressing exceptions. On I/O error, this silently loses data, but data loss must never be silent. If you write a library, just let the caller deal with exceptions.

Reply:
Okay, you mean throw an I/O exception?

Reply:
No. You don't need to throw anything. Exceptions are already being thrown, but your code catches and discards them.